### DISCUSSION TITLE
What is the Ceiling for File Attachments?

### DISCUSSION THREAD

**Discussion by 1artist** [keepassxreboot/keepassxc#9717] [Q&A] [answered]
Greetings,
I am running the latest KeePassXC app and I tried to import into my data base a 19MB Zip file containing scans of some important documents. I did it using the Advance Tab -> Attachments tool. This resulted in KeePassXC flashing me a warning about my data base running slow due to adding such big files to the data base.  So I terminated the operation. 

Can anybody in the Community please comment if importing a 19MB zip file into KeePassXC poses any risk of the data base getting corrupt to the point where it breaks and then I end up locked out my KeePassXC app? That would definitely be a nightmare scenario.  :-)

And what is the allowable file size ceiling KeepassXC is designed to handle as attachments?  Lastly, can anybody recommend a data base app designed with a GUI similar to KeePassXC but that is designed to handle large file sizes as attachments?  Any info will be greatly appreciated.  Thank you for your time.

**Comment by droidmonkey** [comment, marked as answer]
We are not a file storage service, recommend using cryptomator for files. Your entire database is read into memory and decrypted and then encrypted and written to disk. Large files must be processed regardless if they are used or not. That is incredibly inefficient. This is a result of the kdbx format and not something unique to keepassxc.

**Reply by 1artist** [reply]
Droidmonkey, I hope you or somebody in this forum can clarify one more thing about Yubikey the KeepassXC online documentation does not address. I understand Yubikey is used as a 2nd layer of authentication after manually typing in the KDBX password. However, can that functionality be extended so Yubikey automatically types in a long password as well?  In short, I am wondering how typing a very long password on a smartphone keyboard can be made a less tedious task? If it can't be made less tediuos using Yubikey is there some other option that can be used to avoid typing in a very long password using a smartphone keyboard?  Any info welcome.

**Reply by phoerious** [reply]
A YubiKey in KeePassXC is not really a second factor. See: https://keepassxc.org/docs/#faq-yubikey-2fa

There is no point in letting the key enter an additional password. If you want to make your database password-less, just don't define a password and use only the YubiKey.

**Reply by droidmonkey** [reply]
If you mean using yubikey as a password transfer tool between PC and mobile, that is not a function of yubikey. You should instead sync your database or a subset of your database with your mobile device and use a keepass compatible mobile app.

**Reply by 1artist** [reply]
Thank you for your postings. To paint a better picture, I should say my fingers can be compared to fat sausages. For this reason I find it a tedious task to be typing a very long password on my Smartphone's tiny keyboard to get access to my KDBX file. And also for this reason I am very interested in Phoerious' suggestion to go passwordless and just rely on my Yubikey.  However, before doing that I would like to describe how I understand the Yubikey process in the steps I have numbered below: 

1. Mate my Yubikey to my desktop Win10 PC KeepassXC app.
2. Mate my Yubikey to my mobile smartphone KeepassDX app.
3. I then launch KeepassDX on my smartphone, 
4. I then plugin my Yubikey into my smartphone's USB-C port.
5. I then launch my Yubikey Authenticator app and it detects I have my Yubikey plugged into my Smartphone and since it is mated to my KeePassDX app it asks me to physically touch my Yubikey and upon doing that it generates a 6 digit PIN number. 
6. I then copy the PIN and paste it into the KeePassDX password field and this allows me access to my KDBX file without having to manually type a very long password on a tiny smartphone keyboard. 
7. I then close my Yubikey Authenticator app and unmount my Yuikey from the USB-C port. 

Are the steps as I have outlined them for going passwordless and using only Yubikey correct or incorrect?  And more importantly is going passwordless and only using Yubikey less secure, or as secure, or even more secure compared to using a long 40 character password?  The only down side I see is losing your Yubikey. If it gets lost I will be in a world of hurt because of how I will be locked out of my KDBX file.  I think the solution to this problem will be to keep a version of a KDBX file that relies on only a password to access it. 

Droidmonkey, to answer your question I do not think I am talking about using Yubikey as a password transfer tool between my PC and mobile. My goal is to just avoid typing a long password in my Smartphone's tiny keyboard and instead rely on Yubikey to store in its memory a secure key that my KeePassDX app will recognize as valid and thereby grant me access to my KDBX file.  Additionally, I already use Syncthing to keep my KDBX file synced between my desktop Win10 PC and my mobile smartphone. 

Any opinions that will educate me about any making this transition to using Yubikey run more smoothly will be greatly appreciated.

**Reply by michaelk83** [reply]
A 6 digit PIN is not secure on its own. What you're describing sounds like an OTP (temporary One-Time Password) code, which should only ever be used as a 2nd factor. I don't know how KeepassDX works, but in XC you can specify your Yubikey as an additional credential:
https://keepassxc.org/docs/KeePassXC_UserGuide#_opening_an_existing_database (2nd screenshot, "Hardware Key").

You can set this for an existing database in the database settings:
https://keepassxc.org/docs/KeePassXC_UserGuide#_database_settings (step 4, Yubikey Challenge-Response)
This uses the full key data in the hardware key, which is much more secure. Droidmonkey or phoerious can explain the technical details better.
**Make sure to backup your database before you set this up!** If you mess it up and you don't have a backup, you will lose access to all your passwords.

Once you have that set up, then you can drop the main text password ("Remove Password") to go passwordless. I imagine in KeepassDX there is also an option to specify the hardware key as a credential when unlocking the database. You should not need the 3rd party authenticaor app.

**Comment by 1artist** [comment]
My apologies to everybody for confusing my postings between two different Q&A topics I created (#9717 and #9726). This topic titled "What is the Ceiling for File Attachments? #9717" has actually been solved.  I have looked at Cryptomator as Droidmonkey suggested and this is the ideal app to use for storing important documents such as passports and the like. 

Nevertheless, the last five posting made in this thread actually belong in the topic titled "Does KeepassXC support Yubikey selectively across devices? #9726".  Before answering to Michaelk83 comment I will wait to hear back from the board admin with word on whether it is possible to move the last five posts in this topic #9717 over to topic #9726 where they belong.  Again, sorry for messing this up.

**Reply by droidmonkey** [reply]
It is not possible to move posts between discussions.
